src/data/actions/assets.js

```diff
@@ -108,3 +108,52 @@ export const toggleLockAsset = (asset, courseDetails) =>
 export const clearAssetsStatus = () =>
   dispatch =>
     dispatch({ type: assetActions.CLEAR_ASSETS_STATUS });
+
+export const uploadingAssets = count => ({
+  type: assetActions.UPLOADING_ASSETS,
+  count,
+});
+
+export const uploadAssetSuccess = (asset, response) => ({
+  type: assetActions.UPLOAD_ASSET_SUCCESS,
+  asset,
+  response,
+});
+
+export const uploadAssetFailure = (file, response) => ({
+  type: assetActions.UPLOAD_ASSET_FAILURE,
+  file,
+  response,
+});
+
+export const uploadAssets = (files, courseDetails) =>
+  (dispatch, getState) => {
+    dispatch(uploadingAssets(files.length));
+    files.forEach((file) => {
+      clientApi.postUploadAsset(courseDetails.id, file)
+        .then((response) => {
+          if (response.ok) {
+            response.json().then((data) => {
+              dispatch(uploadAssetSuccess(data.asset, response));
+              dispatch(getAssets(getState().request, courseDetails));
```

Comment:
I totally "get" this case! I had thought about using skeleton states, but the issues with this is determining whether the file has been filtered by file type, winnowed down by search, paginated, and sorted. It's difficult to tell where the file should go and if it should even be displayed. I like the idea though and think it's feasible--just would take more effort than I'm wanting to put in at the moment. 😞

Reply:
I had forgotten about the filters and pagination. For what it's worth, I think it would be fine if the uploaded files were inserted at the top of the list until the GET request came back with their actual positions. But yeah, probably not worth the effort for now.